feat(transformer): enable `ArrowFunctionConverter` in `async-to-generator` and `async-generator-functions` plugins

Dunqing opened the pull request:
Part of https://github.com/oxc-project/oxc/pull/7074

In async-to-generator and async-generator-functions plugins, we may need to transform the async arrow function to a regular generator function, now we can reuse the ability of the ArrowFunction plugin by enabling `ArrowFunctionConverter`.

I will fix semantic errors in the follow-up PR

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7113?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7113** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7113?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#7107** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7107?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Dunqing commented on crates/oxc_transformer/src/common/arrow_function_converter.rs:
```diff
@@ -366,6 +380,19 @@ impl<'a> ArrowFunctionConverter<'a> {
+                    if self.is_async_only() && *func.r#async() &&
+                    // Check the function is part of a method definition or object property,
+                    // if it is, then we treat it as a arrow function.
+                    matches!(
+                        ctx.ancestor(index + 1),
+                        // `class A { async foo() {} }` | `({ async foo() {} })`
+                        Ancestor::MethodDefinitionValue(_) | Ancestor::ObjectPropertyValue(_)
+                    ) {
+                        return Some(func.scope_id().get().unwrap())
+                    }
```
We need to check the parent ancestor of the current ancestor here. I didn't thought a better way to do it

overlookmotel replied:
How about this?

```rs
let mut ancestors = ctx.ancestors();
while let Some(ancestor) = ancestors.next() {
    match ancestor {
        // ...
        Ancestor::FunctionBody(func) => {
            if self.is_async_only() && *func.r#async()
            && matches!(
                ancestors.next().unwrap(),
                Ancestor::MethodDefinitionValue(_) | Ancestor::ObjectPropertyValue(_)
            ) {
                return Some(func.scope_id().get().unwrap());
            }
            return None;
        }
    }
}
unreachable!();
```

Dunqing replied:
Looks like better than mine!

overlookmotel commented:
### Merge activity

* **Nov 5, 6:19 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
feat(transformer): enable `ArrowFunctionConverter` in `async-to-generator` and `async-generator-functions` plugins (#7113)

Part of https://github.com/oxc-project/oxc/pull/7074

In async-to-generator and async-generator-functions plugins, we may need to transform the async arrow function to a regular generator function, now we can reuse the ability of the ArrowFunction plugin by enabling `ArrowFunctionConverter`.

I will fix semantic errors in the follow-up PR
